### DISCUSSION TITLE
can cypress cypress be used for API testing

### DISCUSSION THREAD

**Discussion by pj-alves** [cypress-io/cypress#23658] [General]
There are many Ambassadors that are giving workshops on how to use API tests, and they look particularly good. So, I want to explore the option of using Cypress for API testing. 

But I wonder what the company thinks about API testing on the tool. Especially because of the phrase in the official documentation

_"Cypress is mainly designed to run end-to-end and component tests, but if you need to write a few tests that call the backend API using the cy.request command ... who can stop you?""_

It looks that Cypress thinks about API calls in the Cypress ecosystem, as support for UI Tests, and not as a new opportunity to explore the tool for api testing.

Apart from this, there are lots of articles showing that cypress is slow on api testing, because it runs all API tests on a browser. 

I'm wondering if Cypress will invest in this area and remove the need for a browser engine (and that way, speeding tests up)?

**Comment by emilyrohrbough** [comment]
@pj-alves Yes, Cypress can be used to drive API testing. I have read blogs by Cypress users who have done so. 

The Cypress team typically doesn't actively participate in discussion threads unless they are tied to new features or experiments. You'll likely have better luck getting feedback/stimulated a discussion on this topic out in discord: [on.cypress.io/chat](https://on.cypress.io/chat).
